utils/ansible.py

```diff
@@ -0,0 +1,269 @@
+random_token = fauxfactory.gen_alphanumeric(906),
+random_miq_user = fauxfactory.gen_alphanumeric(8)
+local_git_repo = "manageiq_ansible_module"
+pulled_repo_library_path = local_git_repo + "/library"
+remote_git_repo_url = "git://github.com/dkorn/manageiq-ansible-module.git"
+
+
+def fetch_miq_ansible_module():
+    if os.path.isdir(local_git_repo):
+        rmtree(local_git_repo)
+    os.mkdir(local_git_repo)
+    if os.path.isdir(library_path_to_copy_to):
+        rmtree(library_path_to_copy_to)
+    os.mkdir(library_path_to_copy_to)
+    Repo.clone_from(remote_git_repo_url, local_git_repo)
+    src_files = os.listdir(pulled_repo_library_path)
+    for file_name in src_files:
+        full_file_name = os.path.join(pulled_repo_library_path, file_name)
+        if os.path.isfile(full_file_name):
+            copy(full_file_name, library_path_to_copy_to)
+    rmtree(local_git_repo)
+
+
+def get_values_from_conf(provider, script_type):
+    if script_type == 'providers':
+        return {
+            'name': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('name', []),
+            'state': 'present',
+            'miq_url': config_formatter(),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+            'provider_api_hostname': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('hostname', []),
+            'provider_api_auth_token': get_crud('openshift-3').credentials['token'].token,
+            'hawkular_hostname': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('hostname', [])
+        }
+    elif script_type == 'users':
+        return {
+            'fullname': 'MIQUser',
+            'name': 'MIQU',
+            'password': 'smartvm',
+            'state': 'present',
+            'miq_url': config_formatter(),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+        }
+    elif script_type == 'custom_attributes':
+        return {
+            'entity_type': 'provider',
+            'entity_name': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('name', []),
+            'miq_url': config_formatter(),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+        }
+
+
+# TODO Avoid reading files every time
+def read_yml(script, value):
+    with open(yml_path + script + ".yml", 'r') as f:
+            doc = yaml.load(f)
+    return doc[0]['tasks'][0]['manageiq_provider'][value]
+
+
+def get_yml_value(script, value):
+    with open(basic_yml_path + script + ".yml", 'r') as f:
+            doc = yaml.load(f)
+    return doc[0]['tasks'][0]['manageiq_provider'][value]
+
+
+def setup_basic_script(provider, script_type):
+    copyfile(basic_yml_path + script_type + "_basic_script.yml",
```

Comment:
REQUIRED: please use os.path.join as I mentioned before and use variables if you need to use paths several times